synthma/condition_param.go

```diff
@@ -0,0 +1,65 @@
+package synthma
+
+import (
+	"errors"
+
+	"github.com/intervention-engine/fhir/models"
+	"github.com/intervention-engine/fhir/search"
+	"gopkg.in/mgo.v2/bson"
+)
+
+func init() {
+	// Register the condition-code parameter
+	search.GlobalRegistry().RegisterParameterInfo(ConditionCodeParamInfo)
+	search.GlobalRegistry().RegisterParameterParser(ConditionCodeParamInfo.Type, ConditionCodeParamParser)
+	search.GlobalMongoRegistry().RegisterBSONBuilder(ConditionCodeParamInfo.Type, ConditionCodeBSONBuilder)
+}
+
+// ConditionCodeParam represents the condition-code search parameter. Patient's may be searched
+// by conditions they have using a combination of system and code. This behaves exactly the same
+// as a standard TokenParam. See: http://hl7.org/fhir/2016Sep/search.html#token
+type ConditionCodeParam struct {
+	search.TokenParam
+}
+
+// ConditionCodeParamInfo represents the condition-code for Patients' conditions. This allows
+// patients to be searched by conditions they have.
+var ConditionCodeParamInfo = search.SearchParamInfo{
+	Resource: "Patient",
+	Name:     "condition-code",
+	Type:     "condition_code",
```

Comment:
The type must be globally unique.  Although it's not likely someone else is going to register the same type, it's still a good practice to namespace your type.  I'd suggest something like "synthma.patient_condition_code".